File: DynamicKey/AgoraDynamicKey/deno/src/SttTokenBuilder.js
import { AccessToken2 as AccessToken, ServiceRtc, ServiceRtm, ServiceStt } from '../src/AccessToken2.js'
import { Role } from '../src/RtcTokenBuilder2.js'

class SttTokenBuilder {
    /**
     * Builds a Token007 that carries RTC, RTM, and STT services.
     * @param {*} appId The App ID issued to you by Agora.
     * @param {*} appCertificate Certificate of the application that you registered in the Agora Dashboard.
     * @param {*} channelName The unique channel name for the AgoraRTC session in the string format.
     * @param {*} rtcAccount The RTC user's account, max length is 255 bytes.
     * @param {*} rtcRole Role.PUBLISHER for a broadcaster or Role.SUBSCRIBER for an audience member.
     * @param {*} rtcTokenExpire Represented by the number of seconds elapsed since now.
     * @param {*} joinChannelPrivilegeExpire Represented by the number of seconds elapsed since now.
     * @param {*} pubAudioPrivilegeExpire Represented by the number of seconds elapsed since now.
     * @param {*} pubVideoPrivilegeExpire Represented by the number of seconds elapsed since now.
     * @param {*} pubDataStreamPrivilegeExpire Represented by the number of seconds elapsed since now.
     * @param {*} rtmUserId The RTM user's account, max length is 255 bytes.
     * @param {*} rtmTokenExpire Represented by the number of seconds elapsed since now.
     * @return The RTC, RTM, and STT token.
     */
    static buildToken(
        appId,
        appCertificate,
        channelName,
        rtcAccount,
        rtcRole,
        rtcTokenExpire,
        joinChannelPrivilegeExpire,
        pubAudioPrivilegeExpire,
        pubVideoPrivilegeExpire,
        pubDataStreamPrivilegeExpire,
        rtmUserId,
        rtmTokenExpire,
    ) {
        const token = new AccessToken(appId, appCertificate, 0, rtcTokenExpire)

        const serviceRtc = new ServiceRtc(channelName, rtcAccount)
        serviceRtc.add_privilege(ServiceRtc.kPrivilegeJoinChannel, joinChannelPrivilegeExpire)
        if (rtcRole === Role.PUBLISHER) {
            serviceRtc.add_privilege(ServiceRtc.kPrivilegePublishAudioStream, pubAudioPrivilegeExpire)
            serviceRtc.add_privilege(ServiceRtc.kPrivilegePublishVideoStream, pubVideoPrivilegeExpire)
            serviceRtc.add_privilege(ServiceRtc.kPrivilegePublishDataStream, pubDataStreamPrivilegeExpire)
        }
        token.add_service(serviceRtc)

        const serviceRtm = new ServiceRtm(rtmUserId)
        serviceRtm.add_privilege(ServiceRtm.kPrivilegeLogin, rtmTokenExpire)
        token.add_service(serviceRtm)

        token.add_service(new ServiceStt())

        return token.build()
    }
}

export { SttTokenBuilder }

File: DynamicKey/AgoraDynamicKey/rust/examples/stt_token_builder.rs
use agora_token::rtc_token_builder;
use agora_token::stt_token_builder;
use std::env;

fn main() {
    let app_id = env::var("AGORA_APP_ID").unwrap_or_default();
    let app_certificate = env::var("AGORA_APP_CERTIFICATE").unwrap_or_default();

    let channel_name = "stt-channel";
    let rtc_account = "stt-rtc-user";
    let rtc_role = rtc_token_builder::ROLE_PUBLISHER;
    let rtc_token_expire = 3600;
    let join_channel_privilege_expire = 3600;
    let pub_audio_privilege_expire = 3600;
    let pub_video_privilege_expire = 3600;
    let pub_data_stream_privilege_expire = 3600;
    let rtm_user_id = "stt-rtm-user";
    let rtm_token_expire = 3600;

    println!("App Id: {}", app_id);
    if app_id.is_empty() || app_certificate.is_empty() {
        println!("Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE");
        return;
    }

    match stt_token_builder::build_token(
        &app_id,
        &app_certificate,
        channel_name,
        rtc_account,
        rtc_role,
        rtc_token_expire,
        join_channel_privilege_expire,
        pub_audio_privilege_expire,
        pub_video_privilege_expire,
        pub_data_stream_privilege_expire,
        rtm_user_id,
        rtm_token_expire,
    ) {
        Ok(result) => println!("STT token: {}", result),
        Err(err) => println!("{}", err),
    }
}
